File: ws/codena/u3/ws24ConceptsOOP/OOPConcepts/src/main/java/ec/edu/espe/oopconcepts/model/Sheep.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.oopconcepts.model;

import java.util.Date;

/**
 *
 * @author Daniel Codena, CodeBreakers, @ESPE
 */
public class Sheep extends FarmAnimal{
    private Date lastSheering;

    public Sheep(Date lastSheering, int id, String breed, Date bornOnDate, float weight) {
        super(id, breed, bornOnDate, weight);
        this.lastSheering = lastSheering;
    }

    @Override
    public void feed(Food food) {
        System.out.println("Feeding a sheep with --> " + food);
    }

    public Date getLastSheering() {
        return lastSheering;
    }

    public void setLastSheering(Date lastSheering) {
        this.lastSheering = lastSheering;
    }
    
    
}

File: hw/sanchez/u3/hw18OOPConceptsZoo/ZooJavaScript/model/Sheep.js
class Sheep {
  constructor(id, breed, bornOn, weight, lastSheering) {
    this.id = id;
    this.type = 'Sheep';
    this.breed = breed;
    this.bornOn = bornOn;
    this.weight = weight;
    this.lastSheering = lastSheering;
  }

  food(food) {
    console.log(`Feeding sheep with: ${food.description}`);
  }

  toJSON() {
    return {
      id: this.id,
      type: this.type,
      breed: this.breed,
      bornOn: this.bornOn,
      weight: this.weight,
      lastSheering: this.lastSheering
    };
  }
}

module.exports = Sheep;